addons/pos_self_order/models/res_config_settings.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, fields, api
+from werkzeug.urls import url_quote
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    pos_self_order_view_mode = fields.Boolean(
+        related="pos_config_id.self_order_view_mode",
+        readonly=False
+        # compute="_compute_pos_module_pos_self_order", store=True, readonly=False
+    )
+    pos_self_order_image = fields.Image(
+        compute="_compute_pos_module_pos_self_order",
+        store=True,
+        readonly=False,
+        max_width=1920,
+        max_height=1080,
+    )
+    pos_self_order_image_name = fields.Char(
+        compute="_compute_pos_module_pos_self_order", store=True, readonly=False
+    )
+
+    @api.onchange("pos_module_pos_restaurant")
+    def _enable_self_order_view_mode(self):
+        for record in self:
+            record.pos_self_order_view_mode = record.pos_module_pos_restaurant
+
+    # FIXME: the settings don't work correctly:
+    # - disable qr code menu
+    # - save
+    # -  qr code menu is enabled again
```

Comment:
It should work. The problem is with the onchange (`_enable_self_order_view_mode`) above. It always set the value to True. Take inspiration from `pos_is_table_management` to make it work.

Reply:
i've changed it. seems to be working well now

Reply:
i've changed it. seems to be working well now